app/controllers/RejectedNotificationsController.scala

```diff
@@ -37,49 +38,61 @@ class RejectedNotificationsController @Inject() (
   errorHandler: ErrorHandler,
   customsDeclareExportsConnector: CustomsDeclareExportsConnector,
   mcc: MessagesControllerComponents,
+  newErrorReportConfig: NewErrorReportConfig,
+  errorsReportedController: ErrorsReportedController,
   rejectedNotificationPage: rejected_notification_errors
 )(implicit ec: ExecutionContext)
     extends FrontendController(mcc) with I18nSupport with Logging {
 
   def displayPage(id: String): Action[AnyContent] = (authenticate andThen verifyEmail).async { implicit request =>
-    customsDeclareExportsConnector.findDeclaration(id).flatMap {
-      case Some(declaration) =>
-        customsDeclareExportsConnector.findNotifications(id).map { notifications =>
-          val messages = messagesApi.preferred(request).messages
-          val mrn = notifications.headOption.map(_.mrn).getOrElse(messages("rejected.notification.mrn.missing"))
-          Ok(rejectedNotificationPage(None, declaration, mrn, None, getRejectedNotificationErrors(notifications)))
-        }
+    if (newErrorReportConfig.isNewErrorReportEnabled)
+      errorsReportedController.displayPage(id).apply(request)
+    else {
+      customsDeclareExportsConnector.findDeclaration(id).flatMap {
+        case Some(declaration) =>
+          customsDeclareExportsConnector.findNotifications(id).map { notifications =>
+            val messages = messagesApi.preferred(request).messages
+            val mrn = notifications.headOption.map(_.mrn).getOrElse(messages("rejected.notification.mrn.missing"))
+            Ok(rejectedNotificationPage(None, declaration, mrn, None, getRejectedNotificationErrors(notifications)))
+          }
 
-      case _ => errorHandler.internalError(s"Declaration($id) not found for a rejected submission??")
+        case _ => errorHandler.internalError(s"Declaration($id) not found for a rejected submission??")
+      }
     }
   }
 
   def displayPageOnUnacceptedAmendment(actionId: String, draftDeclarationId: Option[String] = None): Action[AnyContent] =
     (authenticate andThen verifyEmail).async { implicit request =>
-      customsDeclareExportsConnector.findAction(actionId).flatMap {
-        case Some(action) =>
-          action.decId match {
-            case Some(decId) =>
-              val declarationId = draftDeclarationId.fold(decId)(identity)
-              customsDeclareExportsConnector.findDeclaration(declarationId).flatMap {
-                case Some(declaration) =>
-                  customsDeclareExportsConnector.findLatestNotification(actionId).map {
-                    case Some(notification) =>
-                      Ok(rejectedNotificationPage(getValue(submissionUuid), declaration, notification.mrn, Some(declarationId), notification.errors))
-                        .addingToSession(submissionActionId -> actionId)
+      if (newErrorReportConfig.isNewErrorReportEnabled)
+        errorsReportedController.displayPageOnUnacceptedAmendment(actionId, draftDeclarationId).apply(request)
```

Comment:
`actionId` and `draftDeclarationId` are not used in `displayPageOnUnacceptedAmendment`